be1-go/hub/standard_hub/mod.go

```diff
@@ -580,3 +678,47 @@ func generateKeys() (kyber.Point, kyber.Scalar) {
 
 	return point, secret
 }
+
+// addMessageId adds a message ID to the map of messageIds by channel of the hub
+func (h *Hub) addMessageId(channelId string, messageId string) {
+	messageIds, channelStored := h.messageIdsByChannel[channelId]
+	if !channelStored {
+		h.messageIdsByChannel[channelId] = append(h.messageIdsByChannel[channelId], messageId)
+	} else {
+		alreadyStored := slices.Contains(messageIds, messageId)
+		if !alreadyStored {
+			h.messageIdsByChannel[channelId] = append(h.messageIdsByChannel[channelId], messageId)
+		}
+	}
+}
+
+// updateRecords updates the hub's globalInbox and messageIdsByChannel to have all the messages ready
+// for heartbeats by locally catching up on channels and store the messages in the hub
+func (h *Hub) updateRecords() {
+	for channelId, channel := range h.channelByID {
+		catchupQuery := method.Catchup{
+			Base: query.Base{
+				JSONRPCBase: jsonrpc.JSONRPCBase{
+					JSONRPC: "2.0",
+				},
+				Method: "catchup",
+			},
+			ID: 0,
+			Params: struct {
+				Channel string "json:\"channel\""
+			}{
+				channelId,
+			},
+		}
+		messages := channel.Catchup(catchupQuery)
+
+		for _, msg := range messages {
+			_, alreadyStored := h.globalInbox.GetMessage(msg.MessageID)
+			if !alreadyStored {
+				h.globalInbox.StoreMessage(msg)
+				h.addMessageId(channelId, msg.MessageID)
+			}
+		}
+	}
+
+}
```

Comment:
I wasn't sure how necessary it was to do that but I was thinking about the messages that are created and sent by the server itself and are only stored at the channel level such as the lao greet. Is there another way to retrieve them ?

Reply:
Update: Actually I'm checking with the scala team and if we only need the messages received by the clients I can just remove it entirely.

Reply:
Update 2: After discussions, it looks like we also need those messages for now but I think we can have this as a temporary solution since we will be optimizing the heartbeat to send ids by batches as oppposed to sending all the ids we know (for examples the latest N messages, to be decided in the protocol).